app/src/main/java/ch/epfl/sweng/erpa/activities/DiceActivity.java

```diff
@@ -1,152 +1,126 @@
-import static ch.epfl.sweng.erpa.util.ActivityUtils.setUsernameInMenu;
+import static ch.epfl.sweng.erpa.util.ActivityUtils.addNavigationMenu;
+import static ch.epfl.sweng.erpa.util.ActivityUtils.onOptionItemSelectedUtils;
+import static ch.epfl.sweng.erpa.util.ActivityUtils.setMenuInToolbar;
 
 public class DiceActivity extends DependencyConfigurationAgnosticActivity {
 
     @Inject UserProfile up;
 
-    Random rng = new Random();    //used as a RNG
-    private final int[] dice = {4, 6, 8, 10, 12, 20, 100};
-    private final int MAX_DICE_NUMBER = 15;
-    private TextView[] allResultViews;
+    private final int MAX_DICE_NUMBER = 9;
+    private final float ROTATION_SPEED = 3f;
+
+    private FlowLayout flowLayout;
+    private ArrayList<DieSketch> allDice = new ArrayList<>();
+
+    /**
+     * Map from die type to filename
+     */
+    private static final Map<Integer, String> compressedDiceResourcesPath =
+            Collections.unmodifiableMap(new HashMap<Integer, String>() {{
+                put(4, "shapes/d4.obj");
+                put(6, "shapes/d6.obj");
+                put(8, "shapes/d8.obj");
+                put(10, "shapes/d10.obj");
+                put(20, "shapes/d20.obj");
+            }});
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_dice);
 
-        allResultViews = getAllResultViews();
-
         //Handle navigationMenu interactions
-        DrawerLayout mDrawerLayout = findViewById(R.id.dice_drawer_layout);
+        addNavigationMenu(this, findViewById(R.id.dice_drawer_layout), findViewById(R.id.dice_navigation_view), up);
+        setMenuInToolbar(this, findViewById(R.id.dice_toolbar));
+        getSupportActionBar().setTitle(R.string.title_dice_activity);
 
-        NavigationView navigationView = findViewById(R.id.dice_navigation_view);
-        navigationView.setNavigationItemSelectedListener(
-                menuItem -> onNavigationItemMenuSelected(menuItem, mDrawerLayout, this));
-        setUsernameInMenu(navigationView, up);
+        flowLayout = findViewById(R.id.dice_layout);
     }
 
-    @OnClick(R.id.rollButton)
-    public void rollDices(View view) {
-        int[] rolls = getNumberOfRolls();
-        if (IntStream.of(rolls).sum() > 15) {
-            createPopup("The number of dice must be less or equal to 15", this);
-        } else {
-            clearAllResults();
-            rollAndShowAllDice(rolls);
-        }
+    //Handle toolbar items clicks
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        Boolean found = onOptionItemSelectedUtils(item.getItemId(), findViewById(R.id.dice_drawer_layout));
+        return found || super.onOptionsItemSelected(item);
     }
 
     /**
-     * Rolls and shows all dice
-     * @param rolls An array containing all the dice to roll
+     * Checks if any die is rolling else rolls all dice
```

Comment:
The description is not very well formulated. ("If no die is rolling, rolls all dice"?)